static/js/analytics.js

```diff
@@ -117,6 +117,156 @@ function openOauth(clickEvt, entrypointElem) {
+
+  const surveyPrompt = document.getElementById("micro-survey-prompt");
+  const surveyType = surveyPrompt.dataset.surveyType;
+  const surveyOptions = document.getElementById("micro-survey-options");
+  switch (surveyType) {
+    case "nps": {
+      const notLikely = document.createElement("li");
+      notLikely.textContent = "Not likely";
+      notLikely.classList = "nps-bookend";
+      surveyOptions.appendChild(notLikely);
+      [...Array(10).keys()].forEach(option => {
+        const li = document.createElement("li");
+        li.classList = "micro-survey-option";
+        li.textContent = option + 1;
+        li.dataset.eventCategory = "NPS Survey";
+        li.dataset.eventAction = "submitted";
+        li.dataset.eventValue = option + 1;
+        if (option < 6) {
+          li.dataset.eventLabel = "detractor";
+          li.dataset.npsValue = -1;
+        } else if (option < 8) {
+          li.dataset.eventLabel = "passive";
+          li.dataset.npsValue = 0;
+        } else {
+          li.dataset.eventLabel = "promoter";
+          li.dataset.npsValue = 1;
+        }
+        li.dataset.ga = "send-ga-funnel-pings";
+        li.addEventListener("click", (evt) => {
+          const eventData = li.dataset;
+          ga("send", "event",
+            eventData.eventCategory,
+            eventData.eventAction,
+            eventData.eventLabel,
+            eventData.eventValue,
+            {
+              dimension1: eventData.eventLabel,
+              metric1: 1,
+              metric2: eventData.eventValue,
+              metric3: eventData.npsValue,
+            }
+          );
+        });
+
+        li.addEventListener("click", setSurveyedCookie);
+        surveyOptions.appendChild(li);
+      });
+      const veryLikely = document.createElement("li");
+      veryLikely.textContent = "Very likely";
+      veryLikely.classList = "nps-bookend";
+      surveyOptions.appendChild(veryLikely);
+    break;
+    }
+    case "pmf": {
+      const options = [
+        "Very disappointed", "Somewhat disappointed", "I wouldn't care"
+      ];
+      options.forEach(option => {
+        const li = document.createElement("li");
+        li.classList = "micro-survey-option";
+        li.textContent = option;
+        li.dataset.eventCategory = "PMF Survey";
+        li.dataset.eventAction = "submitted";
+        li.dataset.eventLabel = option;
+        li.dataset.ga = "send-ga-funnel-pings";
+        li.addEventListener("click", setSurveyedCookie);
+        surveyOptions.appendChild(li);
+      });
+      break;
+    }
+    case "usability":
+    case "credibility":
+    case "appearance": {
+      let countMetric = "metric4";
+      let rankMetric = "metric5";
+      if (surveyType === "credibility") {
+        countMetric = "metric6";
+        rankMetric = "metric7";
+      }
+      if (surveyType === "appearance") {
+        countMetric = "metric8";
+        rankMetric = "metric9";
+      }
```

Comment:
Note: these metric indices are defined in GA:
<img width="409" alt="image" src="https://user-images.githubusercontent.com/71928/108236236-47052d80-710c-11eb-9148-67e0f0850f88.png">